src/DotLiquid.Tests/StandardFilterTests.cs

```diff
@@ -1399,5 +1406,105 @@ public void TestCompact()
+                new { title = "Garlic press", type = "kitchen" }
+            };
+
+            // Check graceful handling of null and empty lists
+            Assert.AreEqual(null, StandardFilters.Where(null, "property", null));
+            CollectionAssert.AreEqual(new string[] { }, StandardFilters.Where(new string[] { }, "property", null));
+
+            // Test filtering by value of a property
+            var expectedKitchenProducts = new[] {
+                new { title = "Spatula", type = "kitchen" },
+                new { title = "Garlic press", type = "kitchen" }
+            };
+            CollectionAssert.AreEqual(expected: expectedKitchenProducts,
+                actual: StandardFilters.Where(products, "type", "kitchen"));
+
+            // Test filtering for existence of a property
+            CollectionAssert.AreEqual(expected: products,
+                actual: StandardFilters.Where(products, "type"));
+
+            // Test filtering for non-existant property
+            var emptyArray = Array.Empty<object>();
+            CollectionAssert.AreEqual(expected: emptyArray,
+                actual: StandardFilters.Where(products, "non_existent_property"));
+        }
+
+        // First sample from specification at https://shopify.github.io/liquid/filters/where/
+        // In this example, assume you have a list of products and you want to show your kitchen products separately.
+        // Using where, you can create an array containing only the products that have a "type" of "kitchen"
+        [Test]
+        public void TestWhere_ShopifySample1()
+        {
+            var products = new[] {
+                new { title = "Vacuum", type = "cleaning" },
+                new { title = "Spatula", type = "kitchen" },
+                new { title = "Television", type = "lounge" },
+                new { title = "Garlic press", type = "kitchen" }
+            };
+
+            // First Shopify sample
+            Helper.AssertTemplateResult(
+                expected: "\r\n\r\nKitchen products:\r\n\r\n- Spatula\r\n\r\n- Garlic press\r\n",
+                template: @"{% assign kitchen_products = products | where: ""type"", ""kitchen"" %}
+
+Kitchen products:
+{% for product in kitchen_products %}
+- {{ product.title }}
+{% endfor %}",
+                localVariables: Hash.FromAnonymousObject(new  { products }) );
+        }
+
+        // Second sample from specification at https://shopify.github.io/liquid/filters/where/
+        // Say instead you have a list of products and you only want to show those that are available to buy.
+        // You can where with a property name but no target value to include all products with a truthy "available" value.
+        [Test]
+        public void TestWhere_ShopifySample2()
+        {
+            var products = new System.Collections.Generic.List<Hash>();
+            products.Add(new Hash { {"title","Coffee mug"}, {"available",true} });

```

Comment:
I believe List <> supports collection initializer, which would simplify the code here.

Reply:
Done